### PR TITLE
Missing postgresql column types (fix #253) 

### LINKED ISSUE #253: (Unsupported SQL column type) Error - EvoMaster process terminated abruptly
Hi, 
I'm trying evomaster on my project, I run evomaster.jar which is downloaded from 
[link](https://github.com/EMResearch/EvoMaster/releases/download/v1.1.0/evomaster.jar) with the command 

> java -jar evomaster.jar - maxTime 150s --outputFolder [directory-name] --outputFormat JAVA_JUNIT_5 

and get the following error

```
* [ERROR] EvoMaster process terminated abruptly. This is likely a bug in EvoMaster. Please copy&paste the following stacktrace, and create a new issue on https://github.com/EMResearch/EvoMaster/issues
java.lang.reflect.InvocationTargetException: invokedynamic: method=public void org.evomaster.core.problem.rest.service.AbstractRestSampler.initialize(), target=org.evomaster.core.problem.rest.service.RestSampler@2f6e28bc
        at com.netflix.governator.lifecycle.LifecycleMethods.methodInvoke(LifecycleMethods.java:316)
        at com.netflix.governator.lifecycle.LifecycleMethods.methodInvoke(LifecycleMethods.java:298)
        at com.netflix.governator.lifecycle.LifecycleManager.startInstance(LifecycleManager.java:266)
        at com.netflix.governator.lifecycle.LifecycleManager.add(LifecycleManager.java:175)
        at com.netflix.governator.guice.InternalLifecycleModule.onProvision(InternalLifecycleModule.java:81)
        at com.google.inject.internal.ProvisionListenerStackCallback$Provision.provision(ProvisionListenerStackCallback.java:120)
        at com.google.inject.internal.ProvisionListenerStackCallback.provision(ProvisionListenerStackCallback.java:66)
        at com.google.inject.internal.ConstructorInjector.construct(ConstructorInjector.java:93)
        at com.google.inject.internal.ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:306)
        at com.google.inject.internal.ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:40)
        at com.google.inject.internal.SingletonScope$1.get(SingletonScope.java:168)
        at com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:39)
        at com.google.inject.internal.FactoryProxy.get(FactoryProxy.java:62)
        at com.google.inject.internal.ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:40)
        at com.google.inject.internal.SingletonScope$1.get(SingletonScope.java:168)
        at com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:39)
        at com.google.inject.internal.InternalInjectorCreator.loadEagerSingletons(InternalInjectorCreator.java:213)
        at com.google.inject.internal.InternalInjectorCreator.injectDynamically(InternalInjectorCreator.java:184)
        at com.google.inject.internal.InternalInjectorCreator.build(InternalInjectorCreator.java:111)
        at com.google.inject.internal.InjectorImpl.createChildInjector(InjectorImpl.java:246)
        at com.netflix.governator.guice.LifecycleInjector.createChildInjector(LifecycleInjector.java:331)
        at com.netflix.governator.guice.LifecycleInjector.createInjector(LifecycleInjector.java:411)
        at com.netflix.governator.guice.LifecycleInjector.createInjector(LifecycleInjector.java:352)
        at org.evomaster.core.Main$Companion.init(Main.kt:236)
        at org.evomaster.core.Main$Companion.initAndRun(Main.kt:139)
        at org.evomaster.core.Main$Companion.main(Main.kt:72)
        at org.evomaster.core.Main.main(Main.kt)
Caused by: java.lang.IllegalArgumentException: Column data type TIME is not supported in Evomaster Data types
        at org.evomaster.core.database.schema.ColumnFactory.parseColumnDataType(ColumnFactory.kt:34)
        at org.evomaster.core.database.schema.ColumnFactory.createColumnFromDto(ColumnFactory.kt:14)
        at org.evomaster.core.database.SqlInsertBuilder.<init>(SqlInsertBuilder.kt:99)
        at org.evomaster.core.problem.rest.service.RestSampler.initSqlInfo(RestSampler.kt:30)
        at org.evomaster.core.problem.rest.service.AbstractRestSampler.initialize(AbstractRestSampler.kt:84)
        at com.netflix.governator.lifecycle.LifecycleMethods.methodInvoke(LifecycleMethods.java:313)
        ... 26 common frames omitted
```

### PR DESCRIPTION
**Pull request by jgaleotti**
Adding support for more postgresql column types. Not all postgresql column types are supported yet. 
The PR includes support for the following postgresql column types:
- MONEY
- BPCHAR
- BYTEA
- INTERVAL
- POINT
- LINE
- LSEG
- BOX
- PATH
- POLYGON
- CIRCLE
- CIDR
- INET
- MACADDR
- MACADDR8
- TSVECTOR
- TSQUERY
- JSONPATH
- INT4RANGE
- INT8RANGE
- NUMRANGE

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -122,7 +122,7 @@
         <testcontainers.version>1.16.0</testcontainers.version>
         <jetbrains.annotations.version>20.1.0</jetbrains.annotations.version>
         <antlr.version>4.9.2</antlr.version>
-        <nlp.version>3.9.2</nlp.version>
+        <nlp.version>4.3.1</nlp.version>
```
Comment (arcuri82) [inline pom.xml:125]: was there any specific reason to bump up this dependency?

**Review by arcuri82 — CHANGES_REQUESTED** [review]

File: client-java/controller-api/src/main/java/org/evomaster/client/java/controller/api/dto/database/schema/DbSchemaDto.java [hunk 2]
```diff
@@ -11,4 +11,5 @@
 
     public List<TableDto> tables = new ArrayList<>();
 
+    public List<EnumeratedTypeDto> postgresEnumeraredTypes = new ArrayList<>();
```
[the remark this answers is not in the document]
Comment (jgaleotti) [reply client-java/controller-api/src/main/java/org/evomaster/client/java/controller/api/dto/database/schema/DbSchemaDto.java:14]: Renamed to enumeratedTypes.

File: core/src/main/kotlin/org/evomaster/core/problem/graphql/GraphQLActionBuilder.kt [hunk 3]
```diff
@@ -75,7 +76,7 @@ object GraphQLActionBuilder {
                 /*
                 In some schemas, "Root" and "QueryType" types define the entry point of the GraphQL query.
                  */
-                if (element.tableType?.toLowerCase() == GqlConst.MUTATION || element.tableType?.toLowerCase() == GqlConst.QUERY || element.tableType?.toLowerCase() == GqlConst.ROOT || element?.tableType?.toLowerCase() == GqlConst.QUERY_TYPE) {
+                if (element.tableType?.lowercase() == GqlConst.MUTATION || element.tableType?.lowercase() == GqlConst.QUERY || element.tableType?.lowercase() == GqlConst.ROOT || element?.tableType?.lowercase() == GqlConst.QUERY_TYPE) {
                     handleOperation(
```
[the remark this answers is not in the document]
Comment (jgaleotti) [reply core/src/main/kotlin/org/evomaster/core/problem/graphql/GraphQLActionBuilder.kt:80]: I will not do such changes in future PRs.

File: core/src/main/kotlin/org/evomaster/core/search/gene/sql/network/SqlCidrGene.kt [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+import org.evomaster.core.search.gene.GeneUtils
+import org.evomaster.core.search.gene.IntegerGene
+import org.evomaster.core.search.service.AdaptiveParameterControl
+import org.evomaster.core.search.service.Randomness
+import org.evomaster.core.search.service.mutator.genemutation.AdditionalGeneMutationInfo
+import org.evomaster.core.search.service.mutator.genemutation.SubsetGeneSelectionStrategy
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+class SqlCidrGene(
```
[the remark this answers is not in the document]
Comment (jgaleotti) [reply core/src/main/kotlin/org/evomaster/core/search/gene/sql/network/SqlCidrGene.kt:15]: Added javadoc to explain what Cidr is.

File: core/src/main/kotlin/org/evomaster/core/search/gene/sql/network/SqlMacAddrGene.kt [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+import org.evomaster.core.search.gene.GeneUtils
+import org.evomaster.core.search.gene.IntegerGene
+import org.evomaster.core.search.service.AdaptiveParameterControl
+import org.evomaster.core.search.service.Randomness
+import org.evomaster.core.search.service.mutator.genemutation.AdditionalGeneMutationInfo
+import org.evomaster.core.search.service.mutator.genemutation.SubsetGeneSelectionStrategy
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+class SqlMacAddrGene(
```
[the remark this answers is not in the document]
Comment (jgaleotti) [reply core/src/main/kotlin/org/evomaster/core/search/gene/sql/network/SqlMacAddrGene.kt:15]: sure

File: pom.xml (hunk 1 above)
Comment (jgaleotti) [reply pom.xml:125]: Some warning issued due to deprected/new release code.

File: core/src/main/kotlin/org/evomaster/core/search/gene/sql/SqlJSONPathGene.kt [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+import org.evomaster.core.search.impact.impactinfocollection.sql.SqlJsonGeneImpact
+import org.evomaster.core.search.service.AdaptiveParameterControl
+import org.evomaster.core.search.service.Randomness
+import org.evomaster.core.search.service.mutator.MutationWeightControl
+import org.evomaster.core.search.service.mutator.genemutation.AdditionalGeneMutationInfo
+import org.evomaster.core.search.service.mutator.genemutation.SubsetGeneSelectionStrategy
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+class SqlJSONPathGene(
```
[the remark this answers is not in the document]
Comment (jgaleotti) [reply core/src/main/kotlin/org/evomaster/core/search/gene/sql/SqlJSONPathGene.kt:18]: done

**Pull request merged**